lib/puppet/functions/defined.rb

```diff
@@ -120,7 +119,6 @@ def is_defined(scope, *vals)
             Puppet::Pops::Evaluator::Runtime3ResourceSupport.find_main_class(scope)
           else
             # Find a resource type, definition or class definition
-            krt = scope.environment.known_resource_types
```

Comment:
@hlindberg @thallgren I'm not sure if these lines have side-effects and are safe to remove? For example, is the expectation that `scope.environment` will raise if the environment doesn't exist?

Reply:
The `scope.environment` should exist - and if it did not it would not be this function's job to detect and report that. There could be a side-effect when calling `known_resource_types`, but again this is very late in that game since we are in a function - could possibly affect tests running with incomplete setup (compared to real runtime), there may also be logic like triggering a reparse etc. which again I think is very late in the game.

Looks like the refactoring of the line after it earlier used `krt` - I think this can be dropped. Ping @thallgren ?